README.md

```diff
@@ -0,0 +1,66 @@
+# go-da
+
+go-da defines a generic Data Availability interface for modular blockchains.
+
+Note that the rollup clients _do not_ need to depend on any implementation,
+they can just make sure that the DA interface is satisfied and start using the
+service. This is a key feature of modular blockchains as they can switch the
+implementations without having to change the interface.
+
+<!-- markdownlint-disable MD013 -->
+[![build-and-test](https://github.com/rollkit/go-da/actions/workflows/ci_release.yml/badge.svg)](https://github.com/rollkit/go-da/actions/workflows/ci_release.yml)
+[![golangci-lint](https://github.com/rollkit/go-da/actions/workflows/lint.yml/badge.svg)](https://github.com/rollkit/go-da/actions/workflows/lint.yml)
+[![Go Report Card](https://goreportcard.com/badge/github.com/rollkit/go-da)](https://goreportcard.com/report/github.com/rollkit/go-da)
+[![codecov](https://codecov.io/gh/rollkit/go-da/branch/main/graph/badge.svg?token=CWGA4RLDS9)](https://codecov.io/gh/rollkit/go-da)
+[![GoDoc](https://godoc.org/github.com/rollkit/go-da?status.svg)](https://godoc.org/github.com/rollkit/go-da)
+<!-- markdownlint-enable MD013 -->
+
+## DA Interface
+
+| Method      | Params                        | Return       |
+| ----------- |-------------------------------| -------------|
+| `MaxBlobSize` |                               | `uint64`       |
+| `Get`         | `ids []ID`                      | `[]Blobs`      |
+| `GetIDs`      | `height uint64`                 | `[]ID`         |
+| `Commit`      | `blobs []Blob`                  | `[]Commitment` |
+| `Validate`    | `ids []Blob, proofs []Proof`    | `[]bool`       |
+
+## Implementations
+
+See [celestia-da](https://github.com/rollkit/celestia-da) for the Celestia
+implementation.
+
```

Comment:
you could mention about:
* the grpc proxy server/client that exist which the implementations can use directly
* DummyDA for testing purpose